s3/client.go

```diff
@@ -168,6 +183,15 @@ func (s *S3Client) CreateBucket(provisionData provider.ProvisionData) error {
 		}
 	}
 	if provisionParams.PublicBucket {
+		logger.Info("delete-public-access-block", lager.Data{"bucket": bucketName})
+		_, err = s.s3Client.DeletePublicAccessBlock(&s3.DeletePublicAccessBlockInput{
```

Comment:
I might have misinterpreted the AWS documentation, but won't you need to PutPublicAccessBlock with RestrictPublicBuckets to false to enable public access. As I understand it the default will be to block it.

Reply:
Quote from https://aws.amazon.com/blogs/aws/heads-up-amazon-s3-security-changes-are-coming-in-april-of-2023/:

If you need public access for a new bucket you can create it as usual and then delete the public access block by calling [DeletePublicAccessBlock](https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeletePublicAccessBlock.html) (you will need s3:PutBucketPublicAccessBlock permission in order to call this function; read [Block Public Access](https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-control-block-public-access.html) to learn more about the functions and the permissions).